modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1161,13 +1165,21 @@ private CompletableFuture<Void> startLocalPartitionsAndClients(
                 boolean shouldStartPartition;
 
                 if (isRecovery) {
-                    // The condition to start the replica is
-                    // `pending.contains(node) || (stable.contains(node) && !pending.isForce())`.
-                    // However we check only the right part of this condition here
-                    // since after `startTables` we have a call to `processAssignmentsOnRecovery`,
-                    // which executes pending assignments update and will start required partitions there.
-                    shouldStartPartition = localMemberAssignmentInStable != null
-                            && (pendingAssignments == null || !pendingAssignments.force());
+                    AssignmentsChain assignmentsChain = assignmentsChains.get(i);
+
+                    if (assignmentsChain == null || assignmentsChain.chain().size() == 1) {
```

Comment:
Let's add comment describing why `assignmentsChain.chain().size() == 1` is a special case and why we have different behaviour instead

Reply:
I would refactor this check to a separate method so we could reuse this method in the second place